### ISSUE TITLE
Translations aren't loaded while isFallback = true

### ISSUE THREAD

**Issue by alex-ragin** [vinissimus/next-translate#575]
Hey!

I noticed an issue that translations aren't loaded while in the isFallback state.

This is my `getStaticPaths` for one of the pages:
```
export const getStaticPaths: GetStaticPaths = async ({ locales }) => {
  const cities = await Api.getCities()
  let paths: Path[] = []

  locales?.forEach((locale) => {
    const localePaths = cities.map((city) => ({
      params: {
        citySlug: city.slug,
      },
      locale,
    }))
    paths = paths.concat(localePaths)
  })

  return {
    paths,
    fallback: true,
  }
```

`getStaticProps`:
```
export const getStaticProps: GetStaticProps<Props, Params> = async ({
  params,
}) => {
  let notFound = false

  if (!params || !params.citySlug) {
    notFound = true
    return {
      notFound,
    }
  }

  const home = await Api.getHome(params.citySlug)
  const city = await Api.getSingleCity(params.citySlug)

  if (!home || !city) {
    notFound = true
  }

  return {
    props: {
      home,
      city,
    },
    notFound,
    revalidate: 60,
  }
}
```

So when the new props are being fetched (isFallback is true), the translations aren't loaded. Once isFallback is false, then you can see all the translations.
![image](https://user-images.githubusercontent.com/22350298/114232720-034cd880-9985-11eb-82e5-29244d87d7a2.png)

![image](https://user-images.githubusercontent.com/22350298/114232679-ec0deb00-9984-11eb-9130-45bb3686f66c.png)

For now, I'm just setting if (isFallback) return null - so that the screen is white while in fallback, but this is not that user friendly :(

Do you guys have any ideas on how to fix that?

**Comment by aralroca** [comment]
I don't know how the isFallback works underneath, so I'll have to investigate. Thank you for reporting it @alex-ragin

**Comment by TomasSestak** [comment]
Happens to me aswell, when using fallback Vercel doesn't let me deploy as the translations outside the page arent loaded

**Comment by aralroca** [comment]
translations are loaded in getStaticProps. I suppose that if you have the fallback, the pages that use the fallback do not load anything (translations or whatever you put inside the getStaticProps), is that so? Or would I have to load it in runtime instead of buildtime? I haven't used the fallback feature yet so I'm not sure how it works. Thanks!

**Comment by svrakata** [comment]
When the fallback is set to true, pages that are not generated at buildtime (not presented as path in getStaticPaths) are produced runtime by calling getStaticProps and added to the other static pages.

**Comment by ztanner** [comment]
We've noticed this problem too and have used the workaround of returning `null` if `isFallback: true`, as suggested above.

I've created a minimal reproduction of this issue here: https://github.com/ztanner/next-translate-fallback-bug, with a description in the README. Hope helps track down the cause - I'll see if I can help investigate as well.

**Comment by berndartmueller** [comment]
I just encountered the same issue after upgrading from 1.0.5 to 1.0.7-canary.1.

**Comment by berndartmueller** [comment]
Digging through the commits since 1.0.5, I was able to pin-point the issue down to https://github.com/vinissimus/next-translate/blob/8abc45855cdc95bed873657379499c758483829a/src/useTranslation.tsx#L11

`useMemo()` has the wrong dependencies. `ctx.t` should be listed in the deps instead of `ctx.lang`.

@aralroca Can you change it and release a hotfix patch version or should I create a PR?

**Comment by aralroca** [comment]
@berndartmueller `ctx.t` is different in every render. The `useMemo` with `lang` dependency was introduced here https://github.com/vinissimus/next-translate/pull/574 to fix this issue https://github.com/vinissimus/next-translate/issues/513#issuecomment-779826418 . Do you think that because of useMemo the isFallback=true does not work?

**Comment by berndartmueller** [comment]
Yes, I changed the code locally and verified that this causes the issue with `isFallback=true`.

Why is `ctx.t` different on each render? `ctx` is a React context and the reference to it and `ctx.t` should be always the same. As long as the context is not manipulated through the provider.

**Comment by berndartmueller** [comment]
I was too hasty with my assumption. It seems this issue exists also before version 1.0.5.

Something is broken when Next.js generates this fallback page (during the initial build). Inspecting the page HTML source of a Next.js page with fallback clearly shows the missing translations. Pages that do not have a fallback (statically generated while building) do have proper translations.

**Comment by berndartmueller** [comment]
I think I found the underlying issue:

See https://nextjs.org/docs/basic-features/data-fetching#fallback-pages

> - The page’s props will be empty.

Even though next-translate loads the needed translations within the webpack loader and puts it into the Next.js  `getStaticProps` `props`, Next.js provides empty page props to the `I18nProvider` in `AppWithTranslations`.

My idea to circumvent this would be checking if `router.isFallback: true` and then loading the translations via `loadNamespaces()` in `AppWithTranslations()`. As this would be async, I'm not really sure yet how to do so.

@aralroca What do you think of this?
